src/lib/mongoBackend/mongoSubscribeContext.cpp

```diff
@@ -183,10 +183,12 @@ HttpStatusCode mongoSubscribeContext
     {
       std::string oidString = oid.toString();
 
-      orion::subCache->insert(new orion::Subscription(tenant, servicePath, requestP, oidString, expiration, notifyFormat));
+      LM_T(LmtMongoSubCache, ("inserting a new sub in cache (%s)", oidString.c_str()));
+      mongoSubCacheItemInsert(tenant.c_str(), servicePath.c_str(), requestP, oidString.c_str(), expiration, throttling, notifyFormat);
     }
 
 
+    reqSemGive(__FUNCTION__, "ngsi10 subscribe request", reqSemTaken);
```

Comment:
Yes, this one definitely makes sense. The sem must protect the subCache also. This is very clear. I moved it on purpose.
That other case, where no subCache calls are ... that one I am not so sure about. The "narrower" the sem is the less other threads have to wait.